charm/src/charm.py

```diff
@@ -25,6 +30,58 @@ def __init__(self, *args: typing.Any) -> None:
         """
         super().__init__(*args)
 
+    def _on_config_changed(self, _event: ops.ConfigChangedEvent) -> None:
+        """"Config changed handler.
+
+        Args:
+            event: the event triggering the handler.
+        """
+        self._copy_files()
+        super()._on_config_changed(_event)
+
+    def _on_django_app_pebble_ready(self, _event: ops.PebbleReadyEvent) -> None:
+        """"Pebble ready handler.
+
+        Args:
+            event: the event triggering the handler.
+        """
+        self._copy_files()
+        super()._on_django_app_pebble_ready(_event)
+
+    def _copy_files(self) -> None:
+        """Copy files needed by git."""
+        container = self.unit.get_container(self._CONTAINER_NAME)
+        if not container.can_connect():
+            self.unit.status = ops.WaitingStatus("Waiting for pebble ready")
+            return
+        if not self.config.get("git_repo"):
+            self.unit.status = ops.WaitingStatus("Config git_repo is required")
```

Comment:
The status message can be overwritten by code execution in the parent class and potentially by another event handler in the parent class. Perhaps move the set warning message part to the `collect-unit-status` and `collect-app-status` event handlers?